client/my-sites/pages/page/index.js

```diff
@@ -57,56 +67,38 @@ const Page = React.createClass( {
 		previousStatus: PropTypes.string,
 		showMoreOptions: PropTypes.bool.isRequired,
 		showPageActions: PropTypes.bool.isRequired,
-	},
-
-	analyticsEvents: {
-		moreOptions: function() {
-			recordEvent( 'Clicked More Options Menu' );
-		},
-		pageTitle: function() {
-			recordEvent( 'Clicked Page Title' );
-		},
-		editPage: function() {
-			recordEvent( 'Clicked Edit Page' );
-		},
-		viewPage: function() {
-			recordEvent( 'Clicked View Page' );
-		},
-		statsPage: function() {
-			recordEvent( 'Clicked Stats Page' );
-		}
-	},
+	}
 
-	togglePageActions: function() {
+	togglePageActions = () => {
 		this.props.togglePageActions();
 		// TODO check previous impl for race conditions
 		if ( this.props.showPageActions ) {
-			this.analyticsEvents.moreOptions();
+			this.props.recordMoreOptions();
 		}
-	},
+	}
 
 	// Construct a link to the Site the page belongs too
-	getSiteDomain: function() {
+	getSiteDomain() {
 		return ( this.props.site && this.props.site.domain ) || '...';
-	},
+	}
 
-	viewPage: function( event ) {
+	viewPage = ( event ) => {
 		event.preventDefault();
 		const { isPreviewable, previewURL } = this.props;
 
 		if ( this.props.page.status && this.props.page.status === 'publish' ) {
-			this.analyticsEvents.viewPage();
+			this.props.recordViewPage();
```

Comment:
Ahh yes, I've hit the exact same warning before actually. Makes sense to leave as is for the sake of being more explicit :)